chinglee-iot/FreeRTOS-Kernel
# Fix coverity violations in FreeRTOS-Kernel.

Fix coverity violations in FreeRTOS-Kernel.

Description
-----------

Fix coverity violations in FreeRTOS-Kernel.

Test Steps
-----------

1. Clone my forked [FreeRTOS repo](https://github.com/ActoryOu/FreeRTOS/tree/kernel-coverity).
1. cd FreeRTOS/FreeRTOS/Test/Coverity
1. Update the compiler configuration in Coverity
`cov-configure --force --compiler cc --comptype gcc`
`cov-configure -co /usr/bin/cc -- -m32 -std=gnu90`
1. Build the (pseudo) application
`cov-build --emit-complementary-info --dir cov-out make`
1. Go to the Coverity output directory (cov-out) and begin Coverity static analysis
`cd cov-out/`
`cov-analyze --dir . --coding-standard-config ../coverity_misra.config`
1. Format the errors in HTML format so that it is more readable while removing the FreeRTOS-Kernel directory from the report
`cov-format-errors --dir . --exclude-files '(/Test/)' --html-output ../output`
1. Then you can get the report at `FreeRTOS/FreeRTOS/Test/Coverity/output`

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
